File: app/src/main/java/com/example/junedshaikh_project/db/ProductDao.java
package com.example.junedshaikh_project.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface ProductDao {
    @Insert
    void insert(Product product);

    @Delete
    void delete(Product product);

    @Query("SELECT * FROM product_table ORDER BY id ASC")
    LiveData<List<Product>> getCartProducts();

    @Query("SELECT Count(*) FROM product_table ")
    LiveData<Integer> getProductCount();
}

File: app/src/main/java/com/example/junedshaikh_project/db/ProductDao.kt
package com.example.junedshaikh_project.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface ProductDao {
    @Insert
    suspend fun insert(product: Product)

    @Delete
    suspend fun delete(product: Product)

    @Query("Select * from product_table order by id ASC")
    fun getCartProducts(): LiveData<List<Product>>
}